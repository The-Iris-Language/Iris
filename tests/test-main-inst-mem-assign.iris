.. Tests assigning a value to a class member variable
class Animal () {
    public:
        int legs;
        int eyes;
        bool fur;
        string color;
        
        string noise(int volume) {
            return "ruh roh raggy";
        }

        void yell() {

            Olympus.print("waahh");
        }

    permit: 
        
        bool alive;
}

class Dog () {
    public:
        int legs1;
    permit: 
        string noise1231231(int volume) {
            return "ruh roh raggy";
        }
}

class Main () {
    public:
       int univ main () {
            Animal a = new Animal();
            a.legs = 0;
            a.color = "brown";
            .. Olympus.print(a.color);
            .. a.yell();
            Dog d = new Dog();
            d.legs = 10;
            .. Olympus.print("Set a.legs to 0");
            return 0;
       }
}