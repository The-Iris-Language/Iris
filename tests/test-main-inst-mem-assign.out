Set a.legs to 0
